webapp/tests/karma/ts/modules/contacts/contacts-content.component.spec.ts

```diff
@@ -240,4 +294,180 @@ describe('Contacts content component', () => {
+        },
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
+      const forms = [
+        { _id: 'form:contact:create:type3' },
+        { _id: 'form:contact:create:type2' },
+      ];
+
+      component.ngOnInit();
+      flush();
+
+      expect(xmlFormsService.subscribe.callCount).to.equal(2);
+      expect(xmlFormsService.subscribe.args[0][0]).to.equal('ContactForms');
+      expect(xmlFormsService.subscribe.args[0][1]).to.deep.equal({ contactForms: true });
+
+      xmlFormsService.subscribe.args[0][2](null, forms);
+
+      expect(globalActions.updateRightActionBar.callCount).to.equal(1);
+      expect(globalActions.updateRightActionBar.args[0][0]).to.deep.equal({
+        childTypes: [
+          {
+            menu_icon: 'fa-building',
+            menu_key: 'Add place',
+            permission: 'can_create_places',
+            types: [
+              {
+                create_form: 'form:contact:create:type2',
+                id: 'type2'
+              },
+              {
+                create_form: 'form:contact:create:type3',
+                id: 'type3'
+              }
+            ]
+          }
+        ]
+      });
+    }));
+
+    it('should filter contact types to allowed ones from selected contact forms', fakeAsync(() => {
+      sinon.resetHistory();
+      contactTypesService.getChildren.resolves([
+        {
+          id: 'type1',
+          create_form: 'form:contact:create:type1',
+        },
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
+      const forms = [
+        { _id: 'form:contact:create:type3', title: 'Type 3', internalId: 3, icon: 'a' },
+        { _id: 'form:contact:create:type2', title: 'Type 2', internalId: 2, icon: 'b' },
+      ];
+
+      component.ngOnInit();
+      flush();
+
+      expect(xmlFormsService.subscribe.callCount).to.equal(2);
+      expect(xmlFormsService.subscribe.args[1][0]).to.equal('ContactList');
+      expect(xmlFormsService.subscribe.args[1][1]).to.deep.equal({
+        contactForms: false,
+        contactSummary: 'test',
+        doc: { _id: 'district-123', phone: '123', muted: true }
+      });
+
+      xmlFormsService.subscribe.args[1][2](null, forms);
+
+      expect(globalActions.updateRightActionBar.callCount).to.equal(1);
+      expect(globalActions.updateRightActionBar.args[0][0]).to.deep.equal({
+        relevantForms: [
+          {
+            code: 2,
+            icon: 'b',
+            showUnmuteModal: true,
+            title: 'Type 2',
+          },
+          {
+            code: 3,
+            icon: 'a',
+            showUnmuteModal: true,
+            title: 'Type 3',
+          }
+        ]
+      });
+    }));
+  });
+
```

Comment:
Can you please add some tests around the action bar when no selectedContact is not defined?

Reply:
Sure, done!